src/wristpy/core/orchestrator.py

```diff
@@ -0,0 +1,259 @@
+"""Python based runner."""
+
+import pathlib
+from typing import Any, Dict, List, Optional
+
+import numpy as np
+import polars as pl
+from pydantic import BaseModel
+
+from wristpy.core import computations, config, models
+from wristpy.io.readers import readers
+from wristpy.processing import analytics, calibration, metrics
+
+
+class InvalidFileTypeError(Exception):
+    """Wristpy cannot save in the given file type."""
+
+    pass
+
+
+class Results(BaseModel):
+    """dataclass containing results of orchestrator.run()."""
+
+    enmo: Optional[models.Measurement] = None
+    anglez: Optional[models.Measurement] = None
+    enmo_epoch1: Optional[models.Measurement] = None
+    anglez_epoch1: Optional[models.Measurement] = None
+    nonwear_array: Optional[models.Measurement] = None
+    sleep_windows: Optional[List[analytics.SleepWindow]] = None
+    physical_activity_levels: Optional[models.Measurement] = None
+    nonwear_epoch1: Optional[models.Measurement] = None
+    sleep_windows_epoch1: Optional[models.Measurement] = None
+
+    def save_results(self, output: pathlib.Path) -> None:
+        """Convert to polars and save the dataframe as a csv or parquet file.
+
+        The data captured by Results can be in one of two temporal resolutions, the
+        unaltered (raw) time stamps taken from the watch data, and the down sampled
+        epoch1 time stamps (5 second intervals). The "raw" time is used in the enmo and
+        anglez data, and the epoch1 time is used for enmo_epoch1, anglez_epoch1,
+        nonwear_epoch1, physical_activity_levels and sleep_windows_epoch1. nonwear_array
+        values are in 15 minute blocks and are upsampled to match epoch1 time for the
+        purposes of saving. Additionally the sleep window data is a list of timestamp
+        pairs, which are used to create a binary array in epoch1 time. Two files will
+        be saved, one of each temporal resolution. This means if output is entered as
+        /path/to/file/file.csv,The file names will be labeled as file_epoch1.csv and
+        file_raw_time.csv in the path/to/file directory.
+
+        Args:
+            output: The path and file name of the data to be saved. as either a csv or
+                parquet files.
+
+        Returns:
+            None.
+
+        Raises:
+            InvalidFileTypeError: If the output file path ends with any extension other
+                than csv or parquet.
+        """
+        if output.suffix not in [".csv", ".parquet"]:
+            raise InvalidFileTypeError(
+                f"The extension: {output.suffix} is not supported.",
+                "Please save the file as .csv or .parquet",
+            )
+
+        results_epoch1_time = self._results_to_dataframe()
```

Comment:
Could it be rewritten as a single function that just takes the field names of the fields that should be saved( e.g. `_result_to_dataframe(["enmo", "anglez", ...])`)

Reply:
See first the comment on epochs at the end of this file. That comment may make this one unnecessary.